llvm/test/Transforms/InstCombine/log-pow.ll

```diff
@@ -12,6 +12,75 @@ define double @log_pow(double %x, double %y) {
   ret double %log
 }
 
+define double @log_powi_const(double %x) {
+; CHECK-LABEL: @log_powi_const(
+; CHECK-NEXT:    [[LOG1:%.*]] = call fast double @llvm.log.f64(double [[X:%.*]])
```

Comment:
This is a bit of a problem, though not one introduced by this PR. The call to log() may set errno in some circumstances, but the call to llvm.log isn't required to do so. If you compile with clang, it will call llvm.log in place of log() when math errno support isn't required. You can get clang to set fast on a call to the log library function using "-ffast-math -fmath-errno" but I'm not sure what that means since the fast-math flags allow us to replace the function.

I'm being pedantic here, and I don't mean for this to block the PR, but this is the primary difference between the library function and the intrinsic, so it seems worth thinking about.

Reply:
@andykaylor Since this is my 2nd PR to llvm, I'm slightly out of my depth here. Are you proposing I drop all the tests with a call to log() directly?

Reply:
The math library function and intrinsic situation is a bit of a mess, and we kind of try to solve the same issue in 2-3 different places. Clang both swaps out the intrinsic for the libcall based on -fno-math-errno, and codegen also tries to specially recognize libcall+readnone callsites as ignore-errno. 

Ideally here we could perform the transform from libcall->libcall and intrinsic->intrinsic, but should avoid mutating between the two

Reply:
> @andykaylor Since this is my 2nd PR to llvm, I'm slightly out of my depth here. Are you proposing I drop all the tests with a call to log() directly?

No. The tests should stay. I was just pointing out that the transformation shouldn't be converting the libcall to an intrinsic, as @arsenm described. That's not something that needs to be changed in the PR though. It's an existing problem.